src/SW_Weather.c

```diff
@@ -1177,13 +1560,81 @@ void _read_weather_hist(
 
 		/* --- Make the assignments ---- */
 		doy--; // base1 -> base0
-        yearWeather->temp_max[doy] = tmpmax;
-        yearWeather->temp_min[doy] = tmpmin;
-        yearWeather->ppt[doy] = ppt;
+        yearWeather->temp_max[doy] = weathInput[maxTempIndex];
+        yearWeather->temp_min[doy] = weathInput[minTempIndex];
+        yearWeather->ppt[doy] = weathInput[precipIndex];
 
         // Calculate average air temperature if min/max not missing
-        if (!missing(tmpmax) && !missing(tmpmin)) {
-          yearWeather->temp_avg[doy] = (tmpmax + tmpmin) / 2.0;
+        if (!missing(weathInput[maxTempIndex]) && !missing(weathInput[minTempIndex])) {
+
+          yearWeather->temp_avg[doy] = (weathInput[maxTempIndex] +
+                                        weathInput[minTempIndex]) / 2.0;
+        }
+
+        if(!weath->use_cloudCoverMonthly && weath->has_cloudCover) {
+            yearWeather->cloudcov_daily[doy] = weathInput[cloudcovIndex];
+        }
+
+        if(!weath->use_windSpeedMonthly) {
+            if(weath->has_sfcWind) {
+                yearWeather->windspeed_daily[doy] = weathInput[windSpeedIndex];
+
+            } else if(weath->has_windComp) {
+                yearWeather->windspeed_daily[doy] = sqrt(squared(weathInput[windComp1Index]) +
+                                                    squared(weathInput[windComp2Index]));
+
+            }
+        }
+
+        if(!weath->use_relHumidityMonthly) {

```

Comment:
`_read_weather_hist()` still does not quite implement calculations and priorities as we need them according to the google document:
- `vp` is currently not (but needs to be) calculated if `huss` is the only humidity input
- `r_humidity_daily` is currently not (but needs to be) calculated if `vp` is the only humidity input
- why is `if(!weath->has_vp) {...} else {...}` placed outside of `if(!weath->use_humidityMonthly)`? `_read_weather_hist()` handles only (and all) calculations from daily inputs

Additionally,
- the condition `weath->has_hurs2 && weath->has_temp2` does not work as intended because temperature values can still be missing even if `weath->has_temp2`; thus, `yearWeather->actualVaporPressure[doy]` needs to be set to missing if any of the temperature values are missing, otherwise, bogus calculations occur
- similarly, if `weath->has_huss`, then `yearWeather->r_humidity_daily[doy]` needs to be set to missing if `yearWeather->temp_avg[doy]` is missing, otherwise, bogus calculations occur